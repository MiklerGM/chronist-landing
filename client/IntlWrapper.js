import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
