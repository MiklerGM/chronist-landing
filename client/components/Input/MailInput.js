import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

class MailInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      valid: true
    };
  }

  handleInvalid(e) {
    e.preventDefault();
    this.setState({ valid: false });
    this.forceUpdate();
  }

  render() {
    return (
      <div className='input--cell'>
        <FormattedMessage id={this.props.placeholder}>
          { (placeholder) => (
            <input
              style={{ gridRow: '1 / 2' }}
              type='email'
              className={(this.state.valid) ? '' : 'invalid'}
              value={this.props.value}
              size='25'
              pattern='^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-z]{2,4}$'
              onInvalid={(e) => this.handleInvalid(e)}
              placeholder={placeholder}
              required
              onChange={(e) => {
                this.setState({ valid: true });
                this.props.cb({ [this.props.name]: e.target.value });
              }}
            />
          )}
        </FormattedMessage>
        <span
          style={{ gridRow: '2 / 3' }}
          className={(this.state.valid)
            ? 'invalid-message__hidden' : 'invalid-message'}
        >
          <span><FormattedMessage id='input.email.invalid' /></span>
        </span>
      </div>
    );
  }
}

MailInput.propTypes = {
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  cb: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired
};

export default MailInput;
