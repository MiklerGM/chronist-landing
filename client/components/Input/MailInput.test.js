import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../../IntlWrapper';
import MailInput from './MailInput';

const TestComponent = () => (
  <IntlWrapper>
    <MailInput
      value=''
      name='email'
      cb={e => console.log('dumb func', e)}
      placeholder='contact.placeholder.email'
    />
  </IntlWrapper>
);

// it must render
it('MailInput render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('MailInput render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('MailInput Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
