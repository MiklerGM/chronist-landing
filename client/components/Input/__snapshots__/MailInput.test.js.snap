// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailInput Snapshot 1`] = `
<div
  className="input--cell"
>
  <input
    className=""
    onChange={[Function]}
    onInvalid={[Function]}
    pattern="^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\\\.[a-z]{2,4}$"
    placeholder="Электронная почта"
    required={true}
    size="25"
    style={
      Object {
        "gridRow": "1 / 2",
      }
    }
    type="email"
    value=""
  />
  <span
    className="invalid-message__hidden"
    style={
      Object {
        "gridRow": "2 / 3",
      }
    }
  >
    <span>
      Неверный формат адреса электронной почты
    </span>
  </span>
</div>
`;
